client/src/templates/Challenges/components/lower-jaw-button.tsx

```diff
@@ -0,0 +1,39 @@
+import React from 'react';
+import { Button } from '@freecodecamp/react-bootstrap';
+import { apiLocation } from '../../../../../config/env.json';
+
+interface LowerJawButtonProps {
+  signed: boolean;
+  buttonText: string;
+  signInText: string;
+  onClick: () => void;
+  challengeIsCompleted: boolean;
+  ref: React.RefObject<HTMLButtonElement>;
+}
+
+export const LowerJawButtons = ({
+  signed,
+  challengeIsCompleted,
+  buttonText,
+  signInText,
+  onClick,
+  ref
+}: LowerJawButtonProps) => {
+  return (
+    <>
+      {!signed && challengeIsCompleted && (
+        <Button block={true} href={`${apiLocation}/signin`} className='btn-cta'>
+          {signInText}
+        </Button>
+      )}
+      <button
+        className={'btn-block btn'}
+        data-cy='lowerJaw-button'
+        onClick={onClick}
+        ref={ref}
+      >
+        {buttonText}
+      </button>
```

Comment:
There is something weird happening, that stops the browsers from focusing on this button, when I use `focus-visible` to set the focus.

Normally browsers are smart enough to do focus on the button, when the function triggers.

Reply:
@Sboonny Can you be a little more specific about what you are having a problem with. I just tested this and it seems to be working fine for me. If the keyboard focus is on the Check Your Code button and you either use the Enter key/Space bar to click the button or use Ctrl + Enter then the focus remains on the button when it switches to Submit. But if the keyboard focus is in the editor and you use Ctrl + Enter then the focus stays in the editor and does not switch to the button, which is how it should be. In other words, if keyboard focus is on the button then it stays on the button, and if keyboard focus is in the editor then it stays in the editor.

Also, the `:focus-visible` pseudo-class doesn't have anything to do with setting focus. It just allows you to display the keyboard focus outline only when the keyboard is actually being used.

Reply:
> Also, the `:focus-visible` pseudo-class doesn't have anything to do with setting focus. It just allows you to display the keyboard focus outline only when the keyboard is actually being used.

I was confused by the below paragraph in this article. https://www.w3.org/WAI/WCAG21/Understanding/focus-visible.html

> “Mode of operation” accounts for user agents which may not always show a focus indicator, or only show the focus indicator when the keyboard is used. User agents may optimise when the focus indicator is shown, such as only showing it when a keyboard is used. Authors are responsible for providing at least one mode of operation where the focus is visible. In most cases there is only one mode of operation so this success criterion applies. The focus indicator must not be time limited, when the keyboard focus is shown it must remain.

I was like, "Event is part of the button, right? So when someone interacts with the event using the keyboard, then they are interacting with the button and will show focus on that button". 🤦 

Reading your comment, I was just hoping it would trigger, I should have created a quick code pen to test this first, instead of assuming. 😅 

---

> @Sboonny Can you be a little more specific about what you are having a problem with. I just tested this and it seems to be working fine for me. If the keyboard focus is on the Check Your Code button and you either use the Enter key/Space bar to click the button or use Ctrl + Enter then the focus remains on the button when it switches to Submit. But if the keyboard focus is in the editor and you use Ctrl + Enter then the focus stays in the editor and does not switch to the button, which is how it should be. In other words, if keyboard focus is on the button then it stays on the button, and if keyboard focus is in the editor then it stays in the editor.

Because of the existence of the tests, I assumed that the accessible behavior, which reminds me of the talk we had about embedding `input` element inside a `button`, which is a WCAG violation and thus not accessible. 

So I really should start looking into component, and see if it's actual accessible and not assume. I should have learned that in that issue 🤦 😥